clients/src/main/java/org/apache/kafka/common/protocol/types/ArrayOf.java

```diff
@@ -72,6 +94,9 @@ public String toString() {
     @Override
     public Object[] validate(Object item) {
         try {
+            if (isNullable() && item == null)
+                return null;
```

Comment:
Apologies, copy and paste error. Its [KAFKA-3604](https://issues.apache.org/jira/browse/KAFKA-3604).